### DISCUSSION TITLE
Allow input components to be able to set the underline input component they use?

### DISCUSSION THREAD

**Discussion by Toshinaki** [mantinedev/mantine#6371] [Feature requests]
I'm trying to create a delyed input with lodash debounce.
And there're `NumberInput`, `PasswordInput`, `Textarea` and `TextInput`.

Now I'm creating a wrapper for all four types of input to implement this.
Basically looks like this:
```
const {value, onChange, ..rest} = props
const [val, setVal] = useState(value)

const handleChange = (e) => {
  setVal(e.target.value)

  _.debounce(() => onChange(val));
}

return <TextInput value={value} onChange={handleChange} />
```
And I'm writing this same logic four times.

It would be great if we can extend `Input` and create one base `DelayedInput`, and pass it as a prop to all types of inputs:
```
<NumberInput input={DelayedInput} />
<PasswordInput input={DelayedInput} />
<Textarea input={DelayedInput} />
<TextInput input={DelayedInput} />
```

Or build this debounce feature into the `Input`, and enable/disable with a prop (`enableDelay`)?

**Comment by rtivital** [comment]
This feature will not work correctly. There are a lot of things that will go wrong with this approach: starting with the inability to handle types correctly and ending with components api differences (some components call onChange function with event object, some call with value).
